mypy/checkexpr.py

```diff
@@ -3583,3 +3592,17 @@ def merge_typevars_in_callables_by_name(
         output.append(target)
 
     return output, variables
+
+
+def is_literal_type_like(t: Optional[Type]) -> bool:
+    """Returns 'true' if the given type context is potentially either a LiteralType,
+    a Union of LiteralType, or something similar.
+    """
+    if t is None:
+        return False
+    elif isinstance(t, LiteralType):
+        return True
+    elif isinstance(t, UnionType):
+        return any(map(is_literal_type_like, t.items))
```

Comment:
Avoid the use of `map()` since mypyc can't compile it efficiently. Use a generator expression instead.